saleor/checkout/utils.py

```diff
@@ -797,33 +797,44 @@ def add_voucher_to_checkout(
         delete_gift_line(checkout_info.checkout, lines)
 
 
-def remove_promo_code_from_checkout(
+def remove_promo_code_from_checkout_or_error(
     checkout_info: "CheckoutInfo", promo_code: str
-) -> bool:
-    """Remove gift card or voucher data from checkout.
+) -> None:
+    """Remove gift card or voucher data from checkout or raise an error."""
 
-    Return information whether promo code was removed.
-    """
     if promo_code_is_voucher(promo_code):
-        return remove_voucher_code_from_checkout(checkout_info, promo_code)
+        remove_voucher_code_from_checkout_or_error(checkout_info, promo_code)
     elif promo_code_is_gift_card(promo_code):
-        return remove_gift_card_code_from_checkout(checkout_info.checkout, promo_code)
-    return False
+        remove_gift_card_code_from_checkout_or_error(checkout_info.checkout, promo_code)
+    else:
+        raise ValidationError(
+            {
+                "promo_code": ValidationError(
+                    "Promo code does not exists.",
+                    code=CheckoutErrorCode.NOT_FOUND.value,
+                )
+            }
+        )
 
 
-def remove_voucher_code_from_checkout(
+def remove_voucher_code_from_checkout_or_error(
     checkout_info: "CheckoutInfo", voucher_code: str
-) -> bool:
-    """Remove voucher data from checkout by code.
+) -> None:
+    """Remove voucher data from checkout by code or raise an error."""
 
-    Return information whether promo code was removed.
-    """
     existing_voucher = checkout_info.voucher
     if existing_voucher and existing_voucher.code == voucher_code:
         remove_voucher_from_checkout(checkout_info.checkout)
         checkout_info.voucher = None
-        return True
-    return False
+    else:
+        raise ValidationError(
+            {
+                "promo_code": ValidationError(
+                    "Cannot remove a voucher not attached to this checkout.",
+                    code=CheckoutErrorCode.VOUCHER_NOT_APPLICABLE.value,
+                )
+            }
+        )
```

Comment:
You shouldn't wrap the `ValidationError` here with the field name reference, as it might be different for different mutations. You should catch this error in mutation and there add a additional layer with the input field name
```suggestion
        raise ValidationError(
                "Cannot remove a voucher not attached to this checkout.",
                code=CheckoutErrorCode.VOUCHER_NOT_APPLICABLE.value,
        )
```